ui/app/src/components/shared/ResourceContextMenu.tsx

```diff
@@ -142,7 +142,7 @@ export const ResourceContextMenu: React.FunctionComponent<ResourceContextMenuPro
   }
 
   // add 'connect' button if we have a URL to connect to
-  if (props.resource.properties.is_exposed_externally === true) {
+  if (props.resource.properties.is_exposed_externally === true && props.resource.properties.connection_uri) {
```

Comment:
'nit' perhaps wrap the exposed_externally conditional in another that checks for connection_uri